src/solver/conv_hip_implicit_gemm_wrw_v4r4_xdlops.cpp

```diff
@@ -0,0 +1,1074 @@
+/*******************************************************************************
```

Comment:
This 1075-lines file has ~90 differences from the existing `conv_hip_implicit_gemm_fwd_v4r4_xdlops.cpp`. Is it possible to generalize existing solver (to serve all directions) instead of copying and editing?

Reply:
I would suggest keeping them as separate code. We tried to reuse the same logic before, and later on found the code need to diverge between fwd and wrw for performance reason, when we further optimized the code

Reply:
Of course, WrW and Fwd diverge. However it is still possible to factor out the common parts, -- this is the matter of some work. I am not insisting however. I just wanted to know have you considered this.

Reply:
[resolved, please do not collapse]

Reply:
For most of the same code, my idea is to create a common base class, subclasses rewrite different code, rather than create a large and complete class. I want to refactor when the software is clear as a whole, not now.